### PR TITLE
Feature/databases

### PR DESCRIPTION
**Pull request by DelroyGayle**
## Description
Created the schemas and mock data for 4 databases

1. User
2. Initiative
3. SchoolStat
4. ScheduleTracker

Modified API.JS  regarding GET endpoints for Initiative, SchoolStat and ScheduleTracker
Mock data supplied so that team can test any current tables and forms

## Related to
#14

## Regarding the following fields

- Programme/Initiative -> reference to programme
- School/NGO name -> reference to school
- No. of new students reached - number
- No. of existing students reached - number
- No. of new teachers reached- number
- No. of existing students reached - number
- No. of tablets (if applicable)- number
- Grades - text (for now, at least 255 chars )
- Languages Taught - text (for now, at least 255 chars )

These are all part of the **SchoolStat** database, not ScheduleTracker
according to my understanding of the supplied tables
Please verify
Thank you

I did **not** add

- No. of new students reached - number
- No. of new teachers reached- number

Because I did not see them in the table supplied
If these need to be added please confirm
Thank you

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -17,6 +27,123 @@ router.get("/", (_, res) => {
...
+				if (reply.length === 0) {
+					genericIntErrorMessage(initiativeId, response, 2, "Initiative");
+					return;
+				}
+
+				return response.json(reply); // Success
+			}
+		}
+	);
+});
```
Comment (berkeli) [inline server/routes/api.js]: Why are you working on initiatives endpoints? this was done by me last week and it isn't in the scope of this ticket

File: server/db/scheduletracker-migration.sql [hunk 2]
```diff
@@ -0,0 +1,1230 @@
+DROP TABLE IF EXISTS scheduletracker;
+-- Create Schedule Tracker Table
+CREATE TABLE scheduletracker (
+   id serial PRIMARY KEY,
+   surveyDate DATE NOT NULL,
```
Comment (MatenRehimi) [inline server/db/scheduletracker-migration.sql]: Remove surveyDate

File: server/db/scheduletracker-migration.sql [hunk 3]
```diff
@@ -0,0 +1,1230 @@
...
+   schoolId INT NOT NULL,
+   deliveredById INT NOT NULL,
```
Comment (MatenRehimi) [inline server/db/scheduletracker-migration.sql]: remove deliveredById

File: server/db/scheduletracker-migration.sql [hunk 4]
```diff
@@ -0,0 +1,1230 @@
...
+   numOfExistingStudents INT NOT NULL,
+   totalNumTablets INT NOT NULL,
+   supportCategory VARCHAR(50) NOT NULL,
+   supportType VARCHAR(50) NOT NULL,
+   programmeInitiativeId INT NOT NULL,
+   grades VARCHAR(255) NOT NULL,
+   languagesTaught VARCHAR(255) NOT NULL,
+   created_ById INT NOT NULL,
+   created_At TIMESTAMP NOT NULL DEFAULT NOW(),
+   FOREIGN KEY (deliveredById)
```
Comment (MatenRehimi) [inline server/db/scheduletracker-migration.sql]: Replace deliveredById with created_ById

File: server/db/schoolstat-migration.sql [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+-- Create School Stat Table
+-- Assume Person '1' did all the entries
+CREATE TABLE schoolstat (
+   id serial PRIMARY KEY,
+   name VARCHAR(50) NOT NULL,
+   location VARCHAR(50) NOT NULL,
+   country VARCHAR(50) NOT NULL,
+   responsibleId INT NOT NULL,
+   status VARCHAR(50) NOT NULL,
+   deploymentYear DATE NOT NULL,
```
Comment (MatenRehimi) [inline server/db/schoolstat-migration.sql]: Replace deploymentYear with deploymentDate

File: server/db/schoolstat-migration.sql [hunk 6]
```diff
@@ -0,0 +1,773 @@
...
+   tabletType VARCHAR(50) NOT NULL,
+   numDoodleStudents INT NOT NULL,
+   internetDetails VARCHAR(50) NOT NULL,
+   totalSchools INT NOT NULL,
+   deployment DATE NOT NULL,
```
Comment (MatenRehimi) [inline server/db/schoolstat-migration.sql]: Remove deployment

File: server/db/schoolstat-migration.sql [hunk 7]
```diff
@@ -0,0 +1,773 @@
+DROP TABLE IF EXISTS schoolstat;
+-- Create School Stat Table
+-- Assume Person '1' did all the entries
+CREATE TABLE schoolstat (
```
Comment (MatenRehimi) [inline server/db/schoolstat-migration.sql]: Check NOT NULL for each field to cope for empty values

Comment (MatenRehimi) [reply server/db/schoolstat-migration.sql]: id to deploymentYear should be NOT NULL

File: server/db/schoolstat-migration.sql [hunk 8]
```diff
@@ -0,0 +1,773 @@
...
+   id serial PRIMARY KEY,
+   name VARCHAR(50) NOT NULL,
+   location VARCHAR(50) NOT NULL,
+   country VARCHAR(50) NOT NULL,
+   responsibleId INT NOT NULL,
+   status VARCHAR(50) NOT NULL,
+   deploymentYear DATE NOT NULL,
+   tabletType VARCHAR(50) NOT NULL,
```
Comment (MatenRehimi) [inline server/db/schoolstat-migration.sql]: tabletType and onwards should allow NULL values

File: server/db/schoolstat-migration.sql [hunk 9]
```diff
@@ -0,0 +1,773 @@
...
+   numDoodleStudents INT NOT NULL,
+   internetDetails VARCHAR(50) NOT NULL,
+   totalSchools INT NOT NULL,
+   deployment DATE NOT NULL,
+   created_ById INT NOT NULL,
```
Comment (MatenRehimi) [inline server/db/schoolstat-migration.sql]: Add foreign key

File: server/routes/api.js [hunk 10]
```diff
@@ -17,6 +27,78 @@ router.get("/", (_, res) => {
...
+
+// GET - Return a single ScheduleTracker by ID.
+router.get("/scheduleTracker/:scheduleTrackerId", function (request, response) {
```
Comment (MatenRehimi) [inline server/routes/api.js]: Remove get as Irina is working on this currently

**Review by berkeli — APPROVED** [review]
Thanks!

**Pull request closed without merging**